### PR TITLE
Allow craft replacements to use groups

### LINKED ISSUE #3368: replacements in minetest.register_craft does not include groups
For example:
```lua
minetest.register_craft({
	type = "shapeless",
	output = "cake:cake",
	recipe = {"farming:flour", "group:water_bucket", "group:food_sugar", "group:food_sugar"},
	replacements = {
		{"group:water_bucket", "bucket:bucket_empty"}
	}
})
```
`group:water_bucket` works just fine for `recipe` but not `replacements`. The only workaround as of now is replacing `group:water_bucket` with `default:bucket_water`, `bucket:bucket_river_water`, and any other water buckets.

### PR DESCRIPTION
**Pull request by TeTpaAka**
Fixes #3368

### CODE REVIEW

File: src/craftdef.cpp [hunk 1]
```diff
@@ -214,11 +214,10 @@ static void craftDecrementOrReplaceInput(CraftInput &input,
 		for (std::vector<std::pair<std::string, std::string> >::iterator
 				j = pairs.begin();
 				j != pairs.end(); ++j) {
-			if (item.name == craftGetItemName(j->first, gamedef)) {
+			if (inputItemMatchesRecipe(item.name, j->first, gamedef->idef())) {
 				if (item.count == 1) {
 					item.deSerialize(j->second, gamedef->idef());
 					found_replacement = true;
-					pairs.erase(j);
```
Comment (est31) [inline src/craftdef.cpp]: This does change behaviour. Before, you had much finer control over what to replace. This would also break the API.

**Comment by est31** [comment]
Other than that: looks good :+1: .

File: src/craftdef.cpp (hunk 1 above)
Comment (TeTpaAka) [inline src/craftdef.cpp]: I reverted it.
I thought, that craft replacements were, that every item type should be replace by another item type, but I agree that you can add multiple replacements to have the same effect.

**Comment by nerzhul** [comment]
:+1:

**Comment by est31** [comment]
3f8eb5e0d0a7dd30bfd6b805d6c03e0f45182e9a

**Pull request closed without merging**